WordPress/openverse
# Render per-repo pull request template

## Fixes




Fixes https://github.com/WordPress/openverse-catalog/issues/671 by @AetherUnbound

## Description




This PR allows the `PULL_REQUEST_TEMPLATE.md` file to be rendered with repo-specific additions. I took inspiration from the existing examples in order to set this one up, but @dhruvkb please let me know if there's a better way to do this! I believe there was also a desire to add some additional checks to the frontend repo's template, that could be done in this PR as well.

In order to make things less cluttered, I also moved all the templates into a `templates` folder at the top level, and changed the logic within the `.pre-commit-config.local.yaml.jinja` file to iterate over a list.

## Testing Instructions



1. `just lint` should produce no changes
1. Running `just render templates/PULL_REQUEST_TEMPLATE.catalog.md.jinja .github/PULL_REQUEST_TEMPLATE.md` should produce a change to the pull request template with the additional rendered content.

## Checklist



      `Update index.md`).
      a parent feature branch.
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin



<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
